plugins/modules/aci_interface_config.py

```diff
@@ -0,0 +1,439 @@
+    spine_access="uni/infra/funcprof/spaccportgrp-{0}",
+    spine_fabric="uni/fabric/funcprof/spportgrp-{0}",
+)
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        policy_group=dict(type="str", aliases=["policy_group_name"]),
+        breakout=dict(type="str", choices=["100g-2x", "100g-4x", "10g-4x", "25g-4x", "50g-8x"]),
+        description=dict(type="str", aliases=["descr"]),
+        node=dict(type="int", aliases=["node_id"]),
+        pc_member=dict(type="str", aliases=["port_channel_member"]),
+        port_type=dict(type="str", default="access", choices=["access", "fabric"]),
+        role=dict(type="str", default="leaf", choices=["leaf", "spine"], aliases=["node_type"]),
+        admin_state=dict(type="str", default="up", choices=["up", "down"]),
+        interface_type=dict(
+            type="str",
+            default="switch_port",
+            choices=["switch_port", "pc_or_vpc", "fc", "fc_port_channel", "leaf_fabric", "spine_access", "spine_fabric"],
+        ),
+        interface=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["role", "node", "interface", "port_type"]],
+            ["state", "present", ["role", "node", "interface", "port_type", "interface_type"]],
+            ["state", "query", ["port_type"]],
+        ],
+        mutually_exclusive=[("policy_group", "breakout")],
+    )
+
+    policy_group = module.params.get("policy_group")
+    breakout = module.params.get("breakout")
+    description = module.params.get("description")
+    node = module.params.get("node")
+    pc_member = module.params.get("pc_member")
+    port_type = module.params.get("port_type")
+    role = module.params.get("role")
+    admin_state = module.params.get("admin_state")
+    interface = module.params.get("interface")
+    interface_type = module.params.get("interface_type")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    if node is not None and node not in range(101, 4001):
+        aci.fail_json(msg="Node ID: {0} is invalid; it must be in the range of 101 to 4000.".format(node))
+
+    card, port_id, sub_port = (None, None, None)
+    if interface is not None:
+        interface_parts = interface.split("/")
+        if len(interface_parts) == 3:
+            card, port_id, sub_port = interface_parts
+        elif len(interface_parts) == 2:
+            card, port_id = interface_parts
+            sub_port = 0
+        else:
+            aci.fail_json(msg="Interface: {0} is invalid; The format must be either card/port/sub_port(1/1/1) or card/port(1/1)".format(interface))
+
+        if card is not None and int(card) not in range(1, 65):
```

Comment:
Should we try/catch the conversion to integer? Think there is a possibility it could lead to value error when incorrect string is provided that is able to split into 2 or 3 parts, like "foo/bar/none".